baybe/campaign.py

```diff
@@ -58,24 +60,42 @@ class Campaign(SerialMixin):
     strategy: Strategy = field(factory=TwoPhaseStrategy)
     """The employed strategy"""
 
-    # Data
-    measurements_exp: pd.DataFrame = field(factory=pd.DataFrame, eq=eq_dataframe)
-    """The experimental representation of the conducted experiments."""
-
-    numerical_measurements_must_be_within_tolerance: bool = field(default=True)
-    """Flag for forcing numerical measurements to be within tolerance."""
-
     # Metadata
-    n_batches_done: int = field(default=0)
+    n_batches_done: int = field(default=0, init=False)
     """The number of already processed batches."""
 
-    n_fits_done: int = field(default=0)
+    n_fits_done: int = field(default=0, init=False)
     """The number of fits already done."""
 
     # Private
-    _cached_recommendation: pd.DataFrame = field(factory=pd.DataFrame, eq=eq_dataframe)
+    _measurements_exp: pd.DataFrame = field(
+        factory=pd.DataFrame, eq=eq_dataframe, init=False
+    )
+    """The experimental representation of the conducted experiments."""
+
+    _cached_recommendation: pd.DataFrame = field(
+        factory=pd.DataFrame, eq=eq_dataframe, init=False
+    )
     """The cached recommendations."""
 
+    # Deprecation
+    numerical_measurements_must_be_within_tolerance: bool = field(default=None)
+    """Deprecated! Raises an error when used."""
+
+    @numerical_measurements_must_be_within_tolerance.validator
+    def _validate_tolerance_flag(self, _, value) -> None:
+        if value is not None:
+            raise DeprecationError(
+                f"Passing a 'numerical_measurements_must_be_within_tolerance' flag to "
```

Comment:
"The" flag instead of "a"

Reply:
I actually think "a" is gucci here. Similarly, if you had a signature `fun(parameter)`, you would also write "Passing a parameter to fun is deprecated" and not "Passing the parameter ..." Will change the other one you mentioned, though!

Reply:
You know what, I'll simply remove both the article AND "flag" in both cases. Makes it even shorter.